lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,484 @@
+        order = {}
+        order[:accountId] = @options[:account_id]
+        order[:referenceCode] = options[:order_id] || generate_unique_id
+        order[:description] = options[:description] || 'unspecified'
+        order[:language] = 'en'
+        post[:transaction][:order] = order
+      end
+
+      def add_buyer(post, options)
+        buyer = {}
+        buyer[:dniNumber] = options[:buyer_dni_number] if options[:buyer_dni_number]
+        buyer[:cnpj] = options[:buyer_cnpj] if options[:buyer_cnpj]
+        buyer[:emailAddress] = options[:email] || 'unspecified@example.com'
+        if address = options[:shipping_address]
+          buyer[:fullName] = address[:name]
+          shipping_address = {}
+          shipping_address[:street1] = address[:address1]
+          shipping_address[:street2] = address[:address2]
+          shipping_address[:city] = address[:city]
+          shipping_address[:state] = address[:state]
+          shipping_address[:country] = address[:country]
+          shipping_address[:postalCode] = address[:zip]
+          shipping_address[:phone] = address[:phone]
+          buyer[:shippingAddress] = shipping_address
+        end
+        post[:transaction][:order][:buyer] = buyer
+      end
+
+      def add_shipping_address(post, options)
+        if address = options[:shipping_address]
+          shipping_address = {}
+          shipping_address[:street1] = address[:address1]
+          shipping_address[:street2] = address[:address2]
+          shipping_address[:city] = address[:city]
+          shipping_address[:state] = address[:state]
+          shipping_address[:country] = address[:country]
+          shipping_address[:postalCode] = address[:zip]
+          shipping_address[:phone] = address[:phone]
+          post[:transaction][:order][:shippingAddress] = shipping_address
+        end
+      end
+
+      def add_invoice(post, money, options)
+        tx_value = {}
+        tx_value[:value] = amount(money)
+        tx_value[:currency] = options[:currency] || currency(money)
+
+        tx_tax = {}
+        tx_tax[:value] = '0'
+        tx_tax[:currency] = options[:currency] || currency(money)
+
+        tx_tax_return_base = {}
+        tx_tax_return_base[:value] = '0'
+        tx_tax_return_base[:currency] = options[:currency] || currency(money)
+
+        additional_values = {}
+        additional_values[:TX_VALUE] = tx_value
+        additional_values[:TX_TAX] = tx_tax
+        additional_values[:TX_TAX_RETURN_BASE] = tx_tax_return_base
+
+        post[:transaction][:order][:additionalValues] = additional_values
+      end
+
+      def add_signature(post)
+        post[:transaction][:order][:signature] = signature_from(post)
+      end
+
+      def signature_from(post)
+        signature_string = [
+          @options[:api_key],
+          @options[:merchant_id],
+          post[:transaction][:order][:referenceCode],
+          post[:transaction][:order][:additionalValues][:TX_VALUE][:value],
```

Comment:
It plays tricks on the eyes, but it's the TX_VALUE ( transaction value? ) that's part of signature. Got rid of all other TAX related values.